### ISSUE TITLE
Consider documenting debug options of doxygen

### ISSUE THREAD

**Issue by marcalff** [doxygen/doxygen#8051]
Doxygen itself contains debug options, great.

Please document in the "-d" debug option as well
- in `doxygen --help`
- in the documentation

Given that -d takes "well known" labels,
please consider something like:
  `doxygen --help --debug`
to dump the content of  table s_labels from file debug.cpp,
so that debug labels can be discovered (without reading the source code).

To reproduce:
`/opt/doxygen-1.8.20/bin/doxygen --help`

**Comment by albert-github** [comment]
At the moment it is already possible to see the "well known" labels by just giving `doxygen -d`, it is a side effect but it gives:
```
error: option "-d" is missing debug specifier.
Developer parameters:
  -m          dump symbol map
  -b          output to wizard
  -T          activates output generation via Django like template
  -d <level>  enable a debug level, such as (multiple invocations of -d are possible):
        cite
        classes
        commentcnv
        commentscan
        extcmd
        filteroutput
        findmembers
        fortranfixed2free
        functions
        lex
        markdown
        plantuml
        preprocessor
        printtree
        time
        validate
        variables
```

**Comment by marcalff** [comment]
Thanks Albert for the clarification.

I tried also
```
malff@localhost.localdomain:bug_artificial> /opt/doxygen-1.8.20/bin/doxygen -d foo
error: option "-d" has unknown debug specifier: "foo".
```
which does not print the list of labels.

It did not occur to me to use -d alone.

**Comment by albert-github** [comment]
Looking at it:
- probably just `doxygen -d` should not give the error message and I don't know whether `exit(1)` or `exit(0) should be used (I now think just `exit(0)` would be OK, with `doxygen --help` also `exit(0)` is used).
- with the `doxygen -d foo` I think it would user friendly to have also the list of possible Developer parameters.

@doxygen what is your opinion?

**Comment by doxygen** [comment]
@albert-github I agree with your improvements; i.e. exit(0) would be better (it is not an error after all) and it more friendly if `-d` with an unknown label produces the list of valid labels.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8052

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
